src/pngwriter.cc

```diff
@@ -45,7 +45,8 @@
 
 // Default Constructor
 ////////////////////////////////////////////////////////////////////////////
-pngwriter::pngwriter()
+pngwriter::pngwriter() :
+  numOmpThreads_( 1 )
```

Comment:
I am not sure if that is the right way to go. By default, the number of OpenMP threads is set via an environment variable, `OMP_NUM_THREADS`.

Therefore, we should not default to `1` but to the value of `OMP_NUM_THREADS` and only if it is not an integer>1 or unset to `1`. (An example how to read env vars is in [libSplash](https://github.com/ComputationalRadiationPhysics/libSplash/blob/e3f2a8802fb16af8e806d71f8ffe04d1e5f2e5e8/src/logging.cpp#L41-L49)).

Note: affects all constructors, so we might want to put this logic in a private member function?

Reply:
@psychocoderHPC maybe you want to initialize it as follows?
```C++
pngwriter::pngwriter()
{
    #pragma omp parallel
    {
        #pragma omp single
        numOmpThreads_ = omp_get_num_threads();
    }

    // ...
}
```